### PR TITLE
Added LocalizedData logic for Write-Verbose and errors to xComputer

### PR DESCRIPTION
**Pull request by lfshr**
Additional work to xComputer to progress the HQRM status and address issue #51

---

This change is [Reviewable](https://reviewable.io/reviews/powershell/xcomputermanagement/53)

### CODE REVIEW

**Comment by PlagueHO** [comment]
Really awesome job @TheFish92 ! Thank you!

I've reviewed this with an eye to the move to HQRM, so there are some things I've requested that although aren't critical to this PR they may need to be done for the move to HQRM.

The reason why we use the ```New-InvalidArgmentExcepton``` or ```New-InvalidOperationException``` calls rather than "Throw" now is that it will create a custom exception object that we can actually test for.

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 7 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/53#-KVTS0oKcS-jvi0S87ln:-KVTS0oLPRznhYJFvKED:b-uuqpw8) ([raw file](https://github.com/powershell/xcomputermanagement/blob/60ec84eb30c18ee21c8dce3f331e610ab44924b8/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L7)):*
> ```PowerShell
> 
> # Forcing en-US for dev purposes
> $LocalizedData = Import-LocalizedData -FileName "MSFT_xComputer.strings.psd1" -UICulture 'en-US'
> ```

Any chance you could grab the localization code that is now being used for PSDSCResources? We're trying to standardize on this method where possible.

See https://github.com/PowerShell/PSDscResources/blob/dev/DscResources/MSFT_GroupResource/MSFT_GroupResource.psm1#L65

No need to define the $LocalizedData as script scope though.

You'll need to copy https://github.com/PowerShell/PSDscResources/blob/dev/DscResources/CommonResourceHelper.psm1 into this module.

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 169 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/53#-KVTTl61RVwPiNaL18XM:-KVTTl61RVwPiNaL18XN:bg6tyel) ([raw file](https://github.com/powershell/xcomputermanagement/blob/60ec84eb30c18ee21c8dce3f331e610ab44924b8/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L169)):*
> ```PowerShell
>         if ($DomainName)
>         {
>             throw ($LocalizedData.SetDomainJoinNoCredential -f $DomainName)
> ```

Rather than "Throwing" an exception like this, can you call ```New-InvalidArgmentExcepton``` or ```New-InvalidOperationException``` found in the CommonResourceHelper.psm1 (that is copied in from https://github.com/PowerShell/PSDscResources/blob/dev/DscResources/CommonResourceHelper.psm1#L24)?

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 245 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/53#-KVTTj4zaFiJvNg1FWbq:-KVTTj4zaFiJvNg1FWbr:bg6tyel) ([raw file](https://github.com/powershell/xcomputermanagement/blob/60ec84eb30c18ee21c8dce3f331e610ab44924b8/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L245)):*
> ```PowerShell
>         if(!($Credential))
>         {
>             throw ($LocalizedData.TestDomainCredentialsNotSpecifiedFailure -f $DomainName)
> ```

Rather than "Throwing" an exception like this, can you call ```New-InvalidArgmentExcepton``` or ```New-InvalidOperationException``` found in the CommonResourceHelper.psm1 (that is copied in from https://github.com/PowerShell/PSDscResources/blob/dev/DscResources/CommonResourceHelper.psm1#L24)?

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 283 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/53#-KVTTq34Amxy9AfCxFUO:-KVTTq35pCWsB9fVB1d4:b4o08ql) ([raw file](https://github.com/powershell/xcomputermanagement/blob/60ec84eb30c18ee21c8dce3f331e610ab44924b8/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L283)):*
> ```PowerShell
>     if ($DomainName -and $WorkGroupName)
>     {
>         throw $LocalizedData.TestDomainOrWorkGroupFailure
> ```

Rather than "Throwing" an exception like this, can you call ```New-InvalidArgmentExcepton``` or ```New-InvalidOperationException``` found in the CommonResourceHelper.psm1 (that is copied in from https://github.com/PowerShell/PSDscResources/blob/dev/DscResources/CommonResourceHelper.psm1#L24)?

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 295 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/53#-KVTSuIOTI8qo4XxbTMy:-KVTSuIOTI8qo4XxbTMz:b-8lbora) ([raw file](https://github.com/powershell/xcomputermanagement/blob/60ec84eb30c18ee21c8dce3f331e610ab44924b8/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L295)):*
> ```PowerShell
>     catch [System.Management.Automation.MethodInvocationException]
>     {
>         Write-Debug $LocalizedData.TestDomainComputerNotMemberOfAny
> ```

Could you change this to a named parameter?

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 403 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/53#-KVTTuJU2zptKFQ1Fmwo:-KVTTuJVQBCCmTfks5Yd:bg6tyel) ([raw file](https://github.com/powershell/xcomputermanagement/blob/60ec84eb30c18ee21c8dce3f331e610ab44924b8/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L403)):*
> ```PowerShell
>     )
> 
>     throw ($LocalizedData.TestNameIsValidFailureError -f $Name, $FailureReason)
> ```

Rather than "Throwing" an exception like this, can you call ```New-InvalidArgmentExcepton``` or ```New-InvalidOperationException``` found in the CommonResourceHelper.psm1 (that is copied in from https://github.com/PowerShell/PSDscResources/blob/dev/DscResources/CommonResourceHelper.psm1#L24)?

---

*[DSCResources/MSFT_xComputer/MSFT_xComputer.psm1, line 424 at r1](https://reviewable.io:443/reviews/powershell/xcomputermanagement/53#-KVTTewc2bCgR3N77Z_y:-KVTTewc2bCgR3N77Z_z:bg6tyel) ([raw file](https://github.com/powershell/xcomputermanagement/blob/60ec84eb30c18ee21c8dce3f331e610ab44924b8/DSCResources/MSFT_xComputer/MSFT_xComputer.psm1#L424)):*
> ```PowerShell
>     if($Throw)
>     {
>         throw $Output
> ```

Rather than "Throwing" an exception like this, can you call ```New-InvalidArgmentExcepton``` or ```New-InvalidOperationException``` found in the CommonResourceHelper.psm1 (that is copied in from https://github.com/PowerShell/PSDscResources/blob/dev/DscResources/CommonResourceHelper.psm1#L24)?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcomputermanagement/53#-:-KVTU0cyJpHtElWcVtKR:b-wx1bvn)*

**Comment by TheFish92** [comment]
@PlagueHO  No problem at all, I'll implement these changes tonight and let you know when I've amended the commit 😄

File: DSCResources/MSFT_xComputer/MSFT_xComputer.psm1 [hunk 1]
```diff
@@ -274,7 +292,7 @@ function GetComputerDomain
     }
     catch [System.Management.Automation.MethodInvocationException]
     {
-        Write-Debug 'This machine is not a domain member.'
+        Write-Debug $LocalizedData.TestDomainComputerNotMemberOfAny
```
Comment (TheFish92) [inline DSCResources/MSFT_xComputer/MSFT_xComputer.psm1]: TODO: Add $DomainName to parameter

**Comment by TheFish92** [comment]
Sorry for the delay on this. Had a horrible coding block last week so decided to stay away from computers at the weekend. I'll finish the changes this week at some point. :)

**Comment by bgelens** [comment]
Assigning abandoned label for now as there has been no response since @lfshr last comment

**Comment by PlagueHO** [comment]
I'm going to close this one as I don't think it is required anymore because most of the changes have been incorporated into other PR's.

**Pull request closed without merging**